ndk/src/midi.rs

```diff
@@ -0,0 +1,271 @@
+//! Bindings for [`AMidiDevice`], [`AMidiInputPort`], and [`AMidiOutputPort`]
+//!
+//! See [the NDK guide](https://developer.android.com/ndk/guides/audio/midi) for
+//! design and usage instructions, and [the NDK reference](https://developer.android.com/ndk/reference/group/midi)
+//! for an API overview.
+//!
+//! [`AMidiDevice`]: https://developer.android.com/ndk/reference/group/midi#amididevice
+//! [`AMidiInputPort`]: https://developer.android.com/ndk/reference/group/midi#amidiinputport
+//! [`AMidiOutputPort`]: https://developer.android.com/ndk/reference/group/midi#amidioutputport
+#![cfg(feature = "media")]
```

Comment:
Shouldn't this live under the `media` module, given that it heavily ties into all its infrastructure and types?

Reply:
I also hesitated to move `midi.rs` to `media`, but since the Android docs describes `AMidi` as a separate module, I left it there. It also seems that `AMidi` does not need `mediasdk` to be linked, so I think it is better to define a new feature "midi" and make `media::{Result, construct, NdkMediaError}` to be compiled when either "media" or "midi" is enabled. But this needs a lot of `#[cfg(any(feature = "media", feature = "midi"))]`s in `media/mod.rs`; would there be a better solution?

Reply:
Great! Since it's before the dawn here, I'll make another commit for this in the morning. Seems like I may have to modify `ndk_sys/src/lib.rs` for `libamidi.so` as well. Thanks for the review!

Reply:
Fixed!

Reply:
Now that I'm at a computer: I meant to look at `hardware_buffer_format.rs` and how I extracted an `enum` from a file that's otherwise guarded by an `api-level` CFG #276. I think we can do a similar thing here, without spreading `cfg(media)` all across `media/mod.rs`?

Reply:
Oh, I got it. Maybe I can add `media_errors.rs` and do similar things for this case as well. I'll make a fix for this.

Reply:
Looks like we're pretty much ready for merge except this point?

Reply:
I'm working on separating `media::*` types that `midi.rs` is dependent on. Maybe we can merge after this.

Reply:
Thanks! Feel free to do so in a separate preliminary PR to make it easier to review / keep things separate.

Let's see if we can do it in a nonbreaking kind of way.

Reply:
Fixed! Still have ugly `#![cfg(any(feature = "media", feature = "midi"))]` at the top of `media_errors.rs`, but `crate::media::{NdkMediaError, Result}` is available via `pub use`:
```rust
// In Line 9 of media/mod.rs:
pub use crate::media_error::{MediaErrorResult, NdkMediaError, Result};
```

Reply:
Ah, I don't think we strictly need to have those `cfg`s. They're only to guard what to link to in `ndk-sys` (and to limit API levels).